code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,439 @@
+			kill()
+		if(istype(temp, /obj/machinery/power/supermatter_crystal/shard))
+			message_admins("found a shard, skipping it")
+			continue
+		sm_crystal += temp
+	var/obj/machinery/power/supermatter_crystal/crystal = pick(sm_crystal)
+	crystal.destabilize()
+
+	priority_announce("WARNING - Detected numerous energy fluctuation generated from your Supermatter; we estimate a [wave_name] of crystal-like creatures \
+						coming from \[REDACTED]; there will be [portal_numbers] portals spread around the station that you must close. Collect the \[REDACTED] \
+						anomaly from the remains of the portals and use it in a crystal stabilizer to stop a ZK-Lambda-Class Cosmic Fragmentation Scenario", "Alert", 'sound/misc/notice1.ogg')
+
+	sleep(10 SECONDS)
+
+	for(var/i = 0, i< portal_numbers, i++)
+		spawn_portal(wave_type)
+
+///Spawn one portal in a random location choosen from the generic_event_spawns list
+/datum/round_event/crystal_invasion/proc/spawn_portal(list/wave_type)
+	var/list/spawners = list()
+	for(var/obj/effect/landmark/event_spawn/temp in GLOB.generic_event_spawns)
+		if(QDELETED(temp))
+			continue
+		if(is_station_level(temp.loc.z))
+			spawners += temp
+
+	if(!spawners.len)
+		message_admins("No APCs on the station, aborting crystal event")
+		return MAP_ERROR
+
+	var/pick_portal = pickweight(wave_type)
+	var/obj/spawner = pick_n_take(spawners)
+	new pick_portal(spawner.loc)
+
+/datum/round_event_control/crystal_invasion/medium
+	name = "Crystal Invasion : Medium"
+	typepath = /datum/round_event/crystal_invasion/medium
+	weight = 6
+	min_players = 25
+	max_occurrences = 3
+	earliest_start = 35 MINUTES
+
+/datum/round_event/crystal_invasion/medium
+	wave_name = "medium wave"
+	portal_numbers = 8
+
+/datum/round_event_control/crystal_invasion/big
+	name = "Crystal Invasion : Big"
+	typepath = /datum/round_event/crystal_invasion/big
+	weight = 9
+	min_players = 35
+	max_occurrences = 3
+	earliest_start = 35 MINUTES
+
+/datum/round_event/crystal_invasion/big
+	wave_name = "big wave"
+	portal_numbers = 11
+
+/datum/round_event_control/crystal_invasion/huge
+	name = "Crystal Invasion : Huge"
+	typepath = /datum/round_event/crystal_invasion/huge
+	weight = 4
+	min_players = 35
+	max_occurrences = 3
+	earliest_start = 45 MINUTES
+
+/datum/round_event/crystal_invasion/huge
+	wave_name = "huge wave"
+	portal_numbers = 17
+
+/*
+This section is for the destabilized SM
+*/
+
+/obj/machinery/destabilized_crystal
+	name = "destabilized crystal"
+	desc = "A strangely translucent and iridescent crystal."
+	icon = 'icons/obj/supermatter.dmi'
+	icon_state = "psy"
+	density = TRUE
+	anchored = TRUE
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1
+	light_range = 4
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	///If not active the crystal will not emit radiations and gases
+	var/active = TRUE
+
+/obj/machinery/destabilized_crystal/Initialize()
+	. = ..()
+	SSshuttle.registerHostileEnvironment(src)
```

Comment:
It seems a bit cruel to keep people from leaving, especially if this event occurs late game and a lot of people are dead for whatever reason.

Reply:
yeah i'll remove it in the next commit

Reply:
even tho it should stay... i mean there is a risk of universe destruction (lore only eh)